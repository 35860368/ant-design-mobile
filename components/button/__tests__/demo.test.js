import { webDemoTest, nativeDemotest } from '../../../tests/shared/demoTest';

webDemoTest('button');
nativeDemotest('button');
