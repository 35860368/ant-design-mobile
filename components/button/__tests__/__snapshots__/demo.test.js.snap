// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/button/demo/basic.md correctly 1`] = `
<div
  class="btn-container need-left-right-margin"
>
  <div>
    <a
      class="btn am-button am-button-primary"
      role="button"
    >
      <span>
        primary 按钮
      </span>
    </a>
    <a
      class="btn am-button am-button-disabled"
      role="button"
    >
      <span>
        disabled 按钮
      </span>
    </a>
    <a
      class="btn am-button am-button-loading am-button-icon"
      role="button"
    >
      <svg
        class="am-icon am-icon-loading am-icon-md"
      >
        <use
          xlink:href="#loading"
        />
      </svg>
      <span>
        loading 按钮
      </span>
    </a>
    <a
      class="btn am-button am-button-icon"
      role="button"
    >
      <svg
        class="am-icon am-icon-check-circle-o am-icon-md"
      >
        <use
          xlink:href="#check-circle-o"
        />
      </svg>
      <span>
        带图标按钮
      </span>
    </a>
    <a
      class="btn am-button am-button-icon"
      role="button"
    >
      <svg
        class="am-icon am-icon-[object Object] am-icon-md"
      >
        <use
          xlink:href="#[object Object]"
        />
      </svg>
      <span>
        本地图标
      </span>
    </a>
    <div
      style="height:0.16rem;"
    />
    <p
      style="margin:30px 0 18px 0;color:#999;"
    >
      inline / small
    </p>
    <div
      class="am-flexbox am-flexbox-align-middle"
      style="margin-bottom:0.16rem;"
    >
      <a
        class="am-button am-button-primary am-button-inline"
        role="button"
        style="margin-right:0.08rem;"
      >
        <span>
          inline
        </span>
      </a>
      <a
        class="am-button am-button-ghost am-button-small am-button-inline"
        role="button"
        style="margin-right:0.08rem;"
      >
        <span>
          inline small
        </span>
      </a>
      <a
        class="am-button am-button-primary am-button-small am-button-inline"
        role="button"
      >
        <span>
          inline small
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`renders ./components/button/demo/basic.tsx correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 80,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "height": 9,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "borderWidth": 1,
              "justifyContent": "center",
            },
            Object {
              "height": 42,
              "paddingLeft": 15,
              "paddingRight": 15,
            },
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#ddd",
            },
            false,
            false,
            false,
            undefined,
          ],
        ]
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 18,
            },
            Object {
              "color": "#000",
            },
            false,
            false,
          ]
        }
      >
        default button
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "borderWidth": 1,
              "justifyContent": "center",
            },
            Object {
              "height": 42,
              "paddingLeft": 15,
              "paddingRight": 15,
            },
            Object {
              "backgroundColor": "#108ee9",
              "borderColor": "#108ee9",
            },
            false,
            false,
            false,
            undefined,
          ],
        ]
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 18,
            },
            Object {
              "color": "#fff",
            },
            false,
            false,
          ]
        }
      >
        primary button
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "borderWidth": 1,
              "justifyContent": "center",
            },
            Object {
              "height": 42,
              "paddingLeft": 15,
              "paddingRight": 15,
            },
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#f4333c",
            },
            false,
            false,
            false,
            undefined,
          ],
        ]
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 18,
            },
            Object {
              "color": "#f4333c",
            },
            false,
            false,
          ]
        }
      >
        warning button
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "borderWidth": 1,
              "justifyContent": "center",
            },
            Object {
              "height": 42,
              "paddingLeft": 15,
              "paddingRight": 15,
            },
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#ddd",
            },
            Object {
              "backgroundColor": "#ddd",
              "borderColor": "#ddd",
            },
            false,
            false,
            undefined,
          ],
        ]
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 18,
            },
            Object {
              "color": "#000",
            },
            Object {
              "color": "#bbb",
            },
            false,
          ]
        }
      >
        disable button
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "borderWidth": 1,
              "justifyContent": "center",
            },
            Object {
              "height": 42,
              "paddingLeft": 15,
              "paddingRight": 15,
            },
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#ddd",
            },
            false,
            false,
            false,
            undefined,
          ],
        ]
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 18,
            },
            Object {
              "color": "#000",
            },
            false,
            false,
          ]
        }
      >
        无点击反馈
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 9,
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "borderWidth": 1,
              "justifyContent": "center",
            },
            Object {
              "height": 42,
              "paddingLeft": 15,
              "paddingRight": 15,
            },
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#ddd",
            },
            false,
            false,
            false,
            undefined,
          ],
        ]
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 18,
            },
            Object {
              "color": "#000",
            },
            false,
            false,
          ]
        }
      >
        自定义点击反馈 style
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 20,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      小按钮
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "borderWidth": 1,
              "justifyContent": "center",
            },
            Object {
              "height": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#108ee9",
            },
            false,
            false,
            false,
            undefined,
          ],
        ]
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 12,
            },
            Object {
              "color": "#108ee9",
            },
            false,
            false,
          ]
        }
      >
        ghost small button
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders ./components/button/demo/complex.md correctly 1`] = `
<div
  class="am-list"
  style="margin:0.1rem 0;background-color:white;"
>
  <div
    class="am-list-body"
  >
    <div
      class="am-list-item am-list-item-middle"
    >
      <div
        class="am-list-line am-list-line-multiple"
      >
        <div
          class="am-list-content"
        >
          区域经理
          <div
            class="am-list-brief"
          >
            可进行收款、退款、折扣管理、查看数据等操作
          </div>
        </div>
        <div
          class="am-list-extra"
        >
          <a
            class="am-button am-button-ghost am-button-small am-button-inline"
            role="button"
          >
            <span>
              small
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="am-list-item am-list-item-middle"
    >
      <div
        class="am-list-line am-list-line-multiple"
      >
        <div
          class="am-list-content"
        >
          区域经理
          <div
            class="am-list-brief"
          >
            可进行收款、退款、折扣管理、查看数据等操作
          </div>
        </div>
        <div
          class="am-list-extra"
        >
          <a
            class="am-button am-button-primary am-button-small am-button-inline"
            role="button"
          >
            <span>
              small
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
